### PR TITLE
Digital Resume Project

### PR DESCRIPTION
**Pull request by ygakkhar-tv**

### CODE REVIEW

File: classes.js [hunk 1]
```diff
@@ -0,0 +1,32 @@
...
+    this.degree = degree;
+    this.fields = fields;
+    this.passingYear = passingYear;
+    this.grades = grades;
+  }
+}
+
+class Skills{
+  constructor(s_kill){
+    this.s_kill = s_kill;
```
Comment (Mr-pm-tv) [inline classes.js]: rename this variable as discussed

File: classes.js [hunk 2]
```diff
@@ -0,0 +1,32 @@
...
+  }
+}
+
+class Certificate{
+  constructor(CField, CAuth, CDate){
+    this.CField = CField;
```
Comment (Mr-pm-tv) [inline classes.js]: rename these variable as discussed

File: classes.js [hunk 3]
```diff
@@ -0,0 +1,32 @@
...
+    this.CAuth = CAuth;
+    this.CDate = CDate;
+  }
+}
+
+class Experience{
+  constructor(sDate, tDate, cName){
```
Comment (Mr-pm-tv) [inline classes.js]: rename these variable as discussed

File: index.js [hunk 4]
```diff
@@ -0,0 +1,171 @@
+//education add and clear button 
+//add button
+let row = 1;
+let education_btn = document.getElementById("add-education");
+let table_body = document.getElementById("table-body");
+
+education_btn.addEventListener("click", () => {
+  let tr = document.createElement("tr");
```
Comment (Mr-pm-tv) [inline index.js:10]: rename this variable as discussed

File: index.js [hunk 5]
```diff
@@ -0,0 +1,171 @@
...
+
+  let template = `
```
Comment (Mr-pm-tv) [inline index.js:13]: rename this variable. we can rename this to educationTemplete

File: index.js [hunk 6]
```diff
@@ -0,0 +1,171 @@
...
+  if (row > 1) {
+    table_body.deleteRow(row - 1);
+    row--;
+  }
+})
+
+
+//skills add and clear button 
+let length=1;
+let skill_count = 1001;
```
Comment (Mr-pm-tv) [inline index.js:36]: remove underscore from variable

File: index.js [hunk 7]
```diff
@@ -0,0 +1,171 @@
...
+let skill_btn = document.getElementById("add_skills");
```
Comment (Mr-pm-tv) [inline index.js:37]: remove underscore from variable

File: index.js [hunk 8]
```diff
@@ -0,0 +1,171 @@
...
+//experience add and clear button 
+//add button
+let flag = 100;
+let experience_btn = document.getElementById("add_experience");
+let experience_block = document.getElementById("experience-data-1");
+
+experience_btn.addEventListener("click", () => {
+  let experienceDiv = document.createElement("div");
+  flag = flag + 1;
+  let template = `
```
Comment (Mr-pm-tv) [inline index.js]: rename this variable to "experienceTeamplete"

File: index.js [hunk 9]
```diff
@@ -0,0 +1,171 @@
...
+
+certificate_btn.addEventListener("click", () => {
+  let certificate_Div = document.createElement("div");
+  count = count + 1;
+  let template = `
+        <input type="date" name="Certification_date" id="Certification_date" class="certificate-date-class"/>
+        <input type="text" name="Certification_feild" id="Certification_feild" class="certificate-field-class"/>
+        <input type="text" name="Certification_authority" id="Certification_authority" class="certificate-auth-class"/>
+        <button id="${count}" onclick="removeCertificate(this)">Clear</button>
+    `;
```
Comment (Mr-pm-tv) [inline index.js]: move this semicolon in line 105

File: index.js [hunk 10]
```diff
@@ -0,0 +1,171 @@
...
+let template_btn2 = document.getElementById("t2");
+template_btn2.addEventListener('click',() => {
+  selected=2;
+  template_btn1.style.backgroundColor = "buttonface";
+  template_btn2.style.backgroundColor = "black";
+});
+
+
+//generate button
+let bool=true;
```
Comment (Mr-pm-tv) [inline index.js]: change this variable name 
For example, if I'm creating a variable to perform the check for the button so I can declare a variable with the name: "isButtonVisible"

File: index.js [hunk 11]
```diff
@@ -0,0 +1,171 @@
+//education add and clear button
```
Comment (ksr877) [inline index.js]: Please follow the camel case for variable name

File: localStorage.js [hunk 12]
```diff
@@ -0,0 +1,143 @@
...
+    educationArr[index].grades = parseInt(eley.value);
+    index++;
+  });
+  localStorage.setItem("educationArr", JSON.stringify(educationArr));
+
+
+//skills
+let noOfSkills = length;
+
+console.log("no Of skills: " +noOfSkills);
```
Comment (Mr-pm-tv) [inline localStorage.js]: remove log's from final code base

File: index.js (hunk 5 above)
Comment (ksr877) [inline index.js:13]: Please use a dynamic way to add table rows, js have insertRow method to add new row

File: localStorage.js [hunk 13]
```diff
@@ -0,0 +1,143 @@
...
+
+  //experience
+  let noOfExperience = document.getElementsByClassName("experience-sdate-class").length;
+  let experienceArr = new Array(noOfExperience);
+
+  for (let i = 0; i < noOfExperience; i++) {
+    experienceArr[i] = new Experience("", "", "");
+  }
+
+  let eIndex = 0;
```
Comment (Mr-pm-tv) [inline localStorage.js]: rename this variable to a relevint name

File: index.js [hunk 14]
```diff
@@ -0,0 +1,171 @@
...
+  selected=2;
+  template_btn1.style.backgroundColor = "buttonface";
+  template_btn2.style.backgroundColor = "black";
+});
+
+
+//generate button
+let bool=true;
+
+let fname_state = true;
```
Comment (Mr-pm-tv) [inline index.js]: variable name!

File: localStorage.js [hunk 15]
```diff
@@ -0,0 +1,143 @@
+//local storage function
+function local_storage() {
+
+  //personal info
+  const Fname = document.getElementById('f-name').value;
+  localStorage.setItem("FName", Fname);
```
Comment (ksr877) [inline localStorage.js]: Please try to set the values in the localStorage using localStorage.setItem('formData',JSON.stringify({fn: firstName.value,
                    ln: lastName.value,
                    emailAddress: emailAddress.value,
                    phone: phone.value,
                    linkedin: linkedin.value,
                    github: github.value,
                    address: address.value,
                    aboutME: aboutME.value
                }));
Only for example

File: localStorage.js [hunk 16]
```diff
@@ -0,0 +1,143 @@
...
+//skills
+let noOfSkills = length;
+
+console.log("no Of skills: " +noOfSkills);
+
+let skillArr = new Array(noOfSkills);
+for (let i = 0; i <noOfSkills; i++) {
+  skillArr[i] = new Skills("");
+}
+let sIndex = 0;
```
Comment (ksr877) [inline localStorage.js]: Use meaningful name for variable

File: localStorage.js (hunk 13 above)
Comment (ksr877) [inline localStorage.js]: Please use menaingful name

File: style.css [hunk 17]
```diff
@@ -0,0 +1,164 @@
...
+.box3 {
+  display: inline-block;
+}
+.per-details {
+  margin-bottom: 20px;
+}
+
+textarea {
+  width: 95%;
+  padding: 2%;
```
Comment (ksr877) [inline style.css:43]: Please always try px insisted of %.

File: style.css [hunk 18]
```diff
@@ -0,0 +1,164 @@
...
+  margin-right: 54px;
+  background: #f2f2f2;
+  border: none;
+}
+#t2 {
+  background: #f2f2f2;
+  border: none;
+}
+#generate-btn {
+  margin-bottom: 2%;
```
Comment (ksr877) [inline style.css]: Please always try px insisted of %.

File: template1.css [hunk 19]
```diff
@@ -0,0 +1,144 @@
...
+  -webkit-print-color-adjust: exact !important;
+}
+.container {
+  margin-left: auto;
+  margin-right: auto;
+}
+.main {
+  display: grid;
+  grid-template-columns: 44% 1% 55%;
+  padding-left: 2%;
```
Comment (ksr877) [inline template1.css:14]: Please always try px insisted of %.

File: template1.css [hunk 20]
```diff
@@ -0,0 +1,144 @@
...
+.education-details {
+  padding-top: 10px;
+}
+
+.edu-1 {
+  padding: 10px 0px 0px 0px;
+}
+
+.edu-2 {
+  padding: 10px 0px 0px 0px;
```
Comment (ksr877) [inline template1.css]: We can use these if you want to apply only one and the other is 0px
padding-top
padding-right
padding-bottom
padding-left

File: template1.css [hunk 21]
```diff
@@ -0,0 +1,144 @@
...
+.templ2 {
+  text-align: center;
+  margin-bottom: 1%;
+}
+#templ2 {
+  background-color: rgb(212, 211, 211);
+  border-radius: 5px;
+  border: none;
+  font-weight: bold;
+  width: 10%;
```
Comment (ksr877) [inline template1.css:141]: Please always try px insisted of %.

File: template2.css [hunk 22]
```diff
@@ -0,0 +1,98 @@
+* {
```
Comment (ksr877) [inline template2.css:1]: margin and padding always default set 0px

File: template2.html [hunk 23]
```diff
@@ -0,0 +1,67 @@
...
+        </div>
+      </div>
+      <div class="main">
+        <div class="half-1">
+          <div class="about margin">
+            <h2>About Me</h2>
+            <p id="aboutme"></p>
+          </div>
+          <hr />
+          <div class="skills margin" id="skill-data">
```
Comment (ksr877) [inline template2.html:39]: skills margin is not valid name pls use valid naming property

File: template2.html [hunk 24]
```diff
@@ -0,0 +1,67 @@
...
+            <h2>Skills</h2>
+          </div>
+          <hr />
+          <div class="exeperince margin">
+            <h2>Certificates</h2>
+            <div class="box flex" id="certificate-data"></div>
```
Comment (ksr877) [inline template2.html:45]: Please use hyphen between class name

File: template2.html [hunk 25]
```diff
@@ -0,0 +1,67 @@
...
+        </div>
+        <div class="half-2">
+          <div class="education">
+            <h2>Education</h2>
+            <div class="box flex" id="education-data"></div>
+          </div>
+          <hr />
+          <div class="work-experience" id="experience-details">
+            <h2>Work Experience</h2>
+            <div class="box flex" id="experience-data"></div>
```
Comment (ksr877) [inline template2.html:56]: Please use hyphen between class name

File: template2.html [hunk 26]
```diff
@@ -0,0 +1,67 @@
...
+          </div>
+        </div>
+      </div>
+    </div>
+
+    <div class="templ1">
```
Comment (ksr877) [inline template2.html]: use meaningful name

File: validations.js [hunk 27]
```diff
@@ -0,0 +1,242 @@
...
+}
+
+//validation function
+function validations() {
+  //valid first name
+  const f_name = document.getElementById("f-name").value;
+  var FName = /^[a-zA-Z]+[a-zA-Z]+$/;
+  if (!f_name.match(FName) || f_name.length < 4) {
+    focus_func(document.getElementById("f-name"));
+    bool = false;
```
Comment (ksr877) [inline validations.js]: use meaningful name for variable

**Review by ksr877 — CHANGES_REQUESTED** [review]

**Review by Mr-pm-tv — CHANGES_REQUESTED** [review]
modify required changes ASAP

File: style.css (hunk 17 above)
Comment (ygakkhar-tv) [reply style.css:43]: tried pixels but % is working much better

File: template1.css (hunk 19 above)
Comment (ygakkhar-tv) [reply template1.css:14]: tried pixels but % is working much better

File: template2.css (hunk 22 above)
Comment (ygakkhar-tv) [reply template2.css:1]: default not zero

File: template2.html (hunk 23 above)
Comment (ygakkhar-tv) [reply template2.html:39]: multiple classes added here

File: template2.html (hunk 24 above)
Comment (ygakkhar-tv) [reply template2.html:45]: multiple classes are used here

**Pull request merged**